lib/customisation.nix

```diff
@@ -306,18 +306,117 @@ rec {
     in if drv == null then null else
       deepSeq drv' drv';
 
-  /* Make a set of packages with a common scope. All packages called
-     with the provided `callPackage` will be evaluated with the same
-     arguments. Any package in the set may depend on any other. The
-     `overrideScope'` function allows subsequent modification of the package
-     set in a consistent way, i.e. all packages in the set will be
-     called with the overridden packages. The package sets may be
-     hierarchical: the packages in the set are called with the scope
-     provided by `newScope` and the set provides a `newScope` attribute
-     which can form the parent scope for later package sets.
+  /**
+    Make an attribute set (a "scope") from functions that take arguments from that same attribute set.
 
-     Type:
-       makeScope :: (AttrSet -> ((AttrSet -> a) | Path) -> AttrSet -> a) -> (AttrSet -> AttrSet) -> AttrSet
+    :::{#ex-makeScope .example}
+    # Create an interdependent package set based on the scope of `pkgs`
```

Comment:
I don't understand what "the scope of `pkgs`" means here. Does it mean something like "will look in `pkgs` second"?

Reply:
Vaguely, yes, technically: all the attributes from the thing returned by `makeScope somethingsomething pkgs` will be available in `self`, such as `callPackage`